src/plugins/data/server/index_patterns/expressions/load_index_pattern.test.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { IndexPatternLoadStartDependencies } from '../../../common/index_patterns/expressions';
+import { createIndexPatternLoad } from './load_index_pattern';
+
+describe('indexPattern expression function', () => {
+  let getStartDependencies: () => Promise<IndexPatternLoadStartDependencies>;
+
+  beforeEach(() => {
+    getStartDependencies = jest.fn().mockResolvedValue({
+      indexPatterns: {
+        get: (id: string) => ({
+          toSpec: () => ({
+            title: 'value',
+          }),
+        }),
+      },
+    });
+  });
+
+  test('returns serialized index pattern', async () => {
+    const indexPatternDefinition = createIndexPatternLoad({ getStartDependencies });
+    const result = await indexPatternDefinition().fn(null, { id: '1' }, {
+      kibanaRequest: {},
+    } as any);
+    expect(result.type).toEqual('index_pattern');
+    expect(result.value.title).toEqual('value');
+  });
+});
```

Comment:
There were already tests for `indexPatternLoad` but I haven't yet added any for `esaggs` -- my plan was to wait until the next PR as I'll be refactoring the args at that time anyway